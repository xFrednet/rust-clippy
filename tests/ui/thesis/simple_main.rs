//@rustc-env: CLIPPY_PRINT_MIR=1
// #[forbid(clippy::borrow_pats)]
use std::collections::HashSet;

struct Bar(u32);

struct Foo {
    field_1: Bar,
    field_2: Bar,
}

#[forbid(clippy::borrow_pats)]
fn example(mut var_1: HashSet<usize>) {
    // Example 1
    var_1.insert(var_1.len());
    
}

fn function(_: &mut Bar, _: &mut Bar) {}


fn main() {}
