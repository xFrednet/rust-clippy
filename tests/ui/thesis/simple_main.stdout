=====
Body for: DefId(0:10 ~ simple_main[8681]::example)
bb0:
    StorageLive(_2)
    StorageLive(_3)
    _3 = &mut _1
    StorageLive(_4)
    StorageLive(_5)
    _5 = &_1
    _4 = std::collections::HashSet::<usize>::len(move _5) -> [return: bb1, unwind: bb4]

bb1:
    StorageDead(_5)
    _2 = std::collections::HashSet::<usize>::insert(move _3, move _4) -> [return: bb2, unwind: bb4]

bb2:
    StorageDead(_4)
    StorageDead(_3)
    StorageDead(_2)
    _0 = const ()
    drop(_1) -> [return: bb3, unwind: bb5]

bb3:
    return

bb4:
    drop(_1) -> [return: bb5, unwind terminate(cleanup)]

bb5:
    resume

=====
[THESIS-ANALYSIS-OUTPUT] {"aggregated_body_stats":{"return_relations_signature":0,"return_relations_found":0,"arg_relations_signature":0,"arg_relations_found":0,"arg_relation_possibly_missed_due_generics":0,"arg_relation_possibly_missed_due_to_late_bounds":0,"ref_stmt_ctn":0,"owned":{"arg_borrow_count":0,"arg_borrow_mut_count":0,"arg_borrow_extended_count":0,"arg_borrow_mut_extended_count":0,"named_borrow_count":0,"named_borrow_mut_count":0,"borrowed_for_closure_count":0,"borrowed_mut_for_closure_count":0,"two_phased_borrows":0}},"body_ctn":3,"total_bb_ctn":8,"total_local_ctn":10,"max_bb_ctn":6,"max_local_ctn":6,"owned_pats":[[{"argument":true,"mutable":true,"owned":true,"copy":false,"drop":"PartDrop","ty":"UserDef"},[],1]],"body_pats":[[{"return_ty":"Unit","is_const":false,"is_safe":true,"is_sync":true,"context":"Free"},[],2],[{"return_ty":"Unit","is_const":false,"is_safe":true,"is_sync":true,"context":"Free"},["NoArguments"],1]],"total_wrap":false}
